recipe/meta.yaml

```diff
@@ -23,39 +23,31 @@ requirements:
     - setuptools
     - setuptools_scm
     - wheel
-    - cython 0.29
-    - numpy   1.19  # [py<310]
-    - numpy   1.21  # [py==310]
-    - numpy   1.23  # [py>=311]
-    - geos
+    - cython >=0.29.24
+    - numpy {{ numpy }}
   run:
     - python
     - {{ pin_compatible('numpy') }}
-    - shapely >=1.6.4
+    - shapely >=1.7
+    - packaging >=20
     - pyshp >=2.1
-    - geos >=3.7.2
-    # including matplotlib-base even though it's optional upstream
-    - matplotlib-base >=3.1
-    - pyproj >=3.0.0
+    - matplotlib-base >=3.4
+    - pyproj >=3.1.0
   run_constrained:
     - scipy >=1.3.1
+    - pillow >=6.1.0
 
 test:
   requires:
     - pip
-    - pytest
+    - pytest >=5.1.2
+    - pytest-xdist
     - scipy >=1.3.1
   commands:
     - python -c "import cartopy.crs as ccrs; import matplotlib.pyplot as plt; ax = plt.axes(projection=ccrs.PlateCarree())"
     - pip check
-    - python -m pytest --pyargs --doctest-modules cartopy  # [not win]
-    # Windows tests fail in several places due to path seperators 
-    #  ie: 
-    # Expected:
-    #   {config[data_dir]}/shapefiles/gshhs/{scale}/GSHHS_{scale}_L{level}.shp
-    # Got:
-    #   {config[data_dir]}\shapefiles\gshhs\{scale}\GSHHS_{scale}_L{level}.shp
-    - python -m pytest --pyargs --doctest-modules cartopy || cmd /K "exit /b 0"  # [win]
+    - pytest -ra -n 4 --pyargs cartopy
+
 about:
   home: https://scitools.org.uk/cartopy
   license: LGPL-3.0-or-later
```

Comment:
The folder `lib/cartopy/data` from the [source dist](https://pypi.org/project/Cartopy/0.22.0/#files) contains a license [file](https://github.com/SciTools/cartopy/blob/v0.22.0/lib/cartopy/data/LICENSE) for data files. Three of them are Public Domain and one is LGPL. So we should add this license file to `license_file` and modify the license as `LGPL-3.0-or-later AND  OGL-UK-1.0 OR OGL-UK-2.0`.
See:
- https://scitools.org.uk/cartopy/docs/latest/copyright.html#cartopy-documentation-and-examples
- https://www.metoffice.gov.uk/hadobs/hadisst/terms_and_conditions.html
- https://www.naturalearthdata.com/about/terms-of-use/
- https://www.earthdata.nasa.gov/learn/use-data/data-use-policy

I'm not sure on why the upstream source contains GPL3 license file https://github.com/SciTools/cartopy/blob/main/COPYING